### ISSUE TITLE
Firejail setup fails with all games on linux

### ISSUE THREAD

**Issue by rcorre** [itchio/itch#2303]
I've installed itch on ArchLinux via the [AUR package](https://aur.archlinux.org/packages/itch/). It launches itch with:

```
#!/bin/sh
itch-setup --prefer-launch -- "@$"
```

When I try to launch any game, the app briefly flickers a dialog that says Firejail is `Preparing`, `Downloading`, and `Ready for Install`, then quickly disappears.

In the logs I see 

```
{"time":1562104612035,"level":30,"msg":"Will perform prereqs installation elevated because (firejail-amd64) has SUID binaries"}
{"time":1562104612035,"level":30,"msg":"→ Invoking self (/home/rcorre/.config/itch/broth/butler/versions/15.12.0/butler)"}
{"time":1562104612035,"level":30,"msg":"  butler ::: --elevate ::: install-prereqs ::: /tmp/butler-prereqs-plan.json358077962"}
{"time":1562104612056,"level":40,"msg":"[err] Error creating textual authentication agent: Error opening current controlling terminal for the process (`/dev/tty'): No such device or address"}
```

Do I need to have some sort of visual sudo prompt enabled? Alternatively, is there a way to have itch just use the `firejail` installed by my package manager that is already on `$PATH`?

**Comment by fasterthanlime** [comment]
> fails with all games

Note that it only needs to be run once per firejail version, not once per game

> Do I need to have some sort of visual sudo prompt enabled?

`butler elevate` does require a working `pkexec`, yes.

Running `pkexec whoami` from a terminal should show a visual password prompt, and entering the password should print `root`. I'm not sure which package exactly provides GTK/QT-based prompts for pkexec, but I'm sure the Arch wiki knows!

> Alternatively, is there a way to have itch just use the `firejail` installed by my package manager that is already on `$PATH`?

Not currently, no. I'm not sure firejail configs are entirely compatible across versions, so we might not want to allow that anyway.

**Comment by rcorre** [comment]
> Running pkexec whoami from a terminal

I verified I had `pkexec` installed but didn't try this. I don't think I have any other GUI programs that ask for elevation, so my `pkexec` might just be busted 😆 

I'll try this tonight and get back to you. Thanks for the pointer!

> I'm not sure firejail configs are entirely compatible across versions, so we might not want to allow that anyway.

I see a number of benefits:

- Eliminate need to ask for elevation from itch, avoid problems with pkexec
- Packagers are already ensuring their distros have up-to-date firejail packages, why [duplicate that effort](https://github.com/itchio/itch/issues/1150)?
- Installing firejail via your package manager should configure it to work out of the box (at least, Arch does, and I'd imagine other distros do too) so itch doesn't have to worry about changing permissions
- Packagers can mark firejail as an (optional) dependency of itch to simplify installation

I can't imagine firejail makes breaking config changes too often, so in my mind the benefits outweigh the presumably rare issue of a config version mismatch.

**Comment by fasterthanlime** [comment]
> Just use the distro's package manager

I'm definitely aware of the benefits, and shipping firejail (and other things) myself is definitely not without its own set of problems, but I'm happy with the call I made years ago.

The "any distro but GNOME/KDE/XFCE running" setup covers a *lot* of our userbase. Older Ubuntus are still used, those definitely have an outdated firejail (so does itch at the moment). I want the sandbox setup to be as frictionless as possible - asking you to open a terminal and run some commands is high friction. Wrapping APT/YUM/pacman/etc. is a hornet's nest (*and* would still require a working visual sudo, so you'd be running into the same issue).

I hope this helps paint the rest of the picture. TL;DR I know, I know - so many possibilities, so little time. Choices had to be made.

**Comment by rcorre** [comment]
`pkexec whoami` failed, and after a lot of digging I found out you have to install a [graphical authentication agent](https://wiki.archlinux.org/index.php/Polkit#Authentication_agents) in addition to `pkexec`.  I installed [polkit-gnome](https://www.archlinux.org/packages/?name=polkit-gnome), started `/usr/lib/polkit-gnome/polkit-gnome-authentication-agent-1`, and then started itch again. This time I got an auth prompt. 

> Wrapping APT/YUM/pacman/etc. is a hornet's nest

To be clear, I wasn't suggesting that itch does anything with your package manager. I was suggesting that package maintainers list firejail as a dependency of itch so installing itch automatically installs firejail through my system's normal process, in the same way that the rest of the deps are installed:

```
[rcorre@midova ~]$ pacman -Qi itch | grep Depends
Depends On      : alsa-lib  libnotify  nss  gconf  gtk2  libxtst  desktop-file-utils  gtk-update-icon-cache  libxss  gtk3
``` 

>  TL;DR I know, I know - so many possibilities, so little time. Choices had to be made.

Fair enough. I did get it working, thanks to your tip, it was just a non-obvious path. Thanks for maintaining this awesome project!
